src/engine_core/src/input/InputManager.hpp

```diff
@@ -0,0 +1,24 @@
+/*! \file InputManager.hpp
+    \author Kyn21kx
+    \date 2024-02-28
+    \brief Receives the input information of the application and makes it available through static methods
+*/
+
+#pragma once
+#include "KeyData.hpp"
+#include <map>
+
+
+class InputManager
+{
+public:
+
+    static bool IsKeyDown(KeyCode key);
+
+    static void SendKeyEvent(KeyCode key, EKeyState state);
+
+private:
+    //TODO: Reserve memory for this map???
+    static std::map<KeyCode, KeyData> s_keyDataByCode;
```

Comment:
While map is easier to use, a switch-case might be more efficient (both in memory and in speed). Although we shouldn't worry about that in the meantime

Reply:
Just a moment to consider in the future.
We should, however, create a way to instrument each function in runtime. Not in the current scope though

Reply:
The purpose behind a map here would be to be able to store previous and current state associated to the KeyCode, not just the current one, but let me know if you have an idea to make that work too